components/listingsPage/__tests__/Filters.test.js

```diff
@@ -0,0 +1,69 @@
+import { shallow } from "enzyme";
+import Filters  from "../filters";
+import { FormDiv, ToggleIcon } from "../utils";
+import { findByTestAtrr } from "../../../Utils/test";
+
+const mockDispatch = jest.fn();
+jest.mock('react-redux', () => ({
+  useSelector: jest.fn(),
+  useDispatch: () => mockDispatch
+}));
+
+jest.mock("react-i18next", () => ({
+    withTranslation: () => (Component) => {
+      Component.defaultProps = { ...Component.defaultProps, t: () => [] };
+      return Component;
+    },
+  }));
+
+
+
+describe("Filters Component Tests", () => {
+
+    let filters;
+    const ToggleMobile = jest.fn();
+      
+        beforeEach(() => {
+            filters = shallow(<Filters onClick={ToggleMobile} t={key => []}/>);
+        });
+      
+        afterEach(() => {
+            filters.unmount();
+        });
```

Comment:
I think we don’t really need to use `beforeEach` and `afterEach` here since  the `<Filters/>` component doesn’t change between tests. Instead we can just define it once as a `const` which may also be a little more efficient. Alternatively we could use `beforeAll` and `afterAll` though with no real setup happening I’m not sure that is necessary either.

Reply:
Its a convention to always clean up after each test to clear any attribute or props that might have been set and also to free space. I can use beforeAll and afterAll if you prefer, i just prefer the each because over time you might add test that changes the component and causing others to fail without realising the cause.

Reply:
My feeling is that where there is no setting up and tearing down to be done it's not really necessary (see documentation [here](https://jestjs.io/docs/en/setup-teardown)). I would say it's a matter of preference so I'm not unhappy with the way you have done it here. I think it would be useful to establish some standards team-wide on this though. 

I do feel the `unmount` method is not appropriate here though (see docs [here](https://enzymejs.github.io/enzyme/docs/api/ShallowWrapper/unmount.html)). I see this as more of a way to mock React native behaviour in tests rather than part of the test itself. Again, I would prefer to expand this into a wider discussion so we can establish good practices and standards.

Reply:
The shallow rendering and unmounting constitute the setting up and tearing down, so I would not have shallow render in each test. Even the [docs](https://enzymejs.github.io/enzyme/docs/api/ShallowWrapper/unmount.html) you referred to called `unmount()` on the `wrapper` after shallow rendering. Unless you want me to do that in every test. But I really want you be happy here :), I am changing them to `beforeAll` and `afterAll`.

Reply:
I'd suggest we widen the discussion to the fuller team before making any changes. I don't feel I have sufficient knowledge of best practices on this. Plus I think we could all benefit from a wider discussion.

Reply:
I think in this case `beforeAll` should be alright, it's not like the structure is going to change in-between these tests which is ultimately what is being checked. And yes, establishing some standards would be nice 👍

Reply:
> The shallow rendering and unmounting constitute the setting up and tearing down, so I would not have shallow render in each test. Even the [docs](https://enzymejs.github.io/enzyme/docs/api/ShallowWrapper/unmount.html) you referred to called `unmount()` on the `wrapper` after shallow rendering. Unless you want me to do that in every test. But I really want you be happy here :), I am changing them to `beforeAll` and `afterAll`.
> 
OK, I think `beforeAll` and `afterAll` is better here but I still think there's a discussion to be had about the `unmount()` method. But for this test let's leave it as it isn't going to cause any problems.